### PR TITLE
Refactor Realm DOM tests to use postMessage to communicate with realm under test

### PR DESCRIPTION
**Pull request by habdelra**
A recent chrome update broke how we were able to take advantage of cross site scripting within different ports of localhost in order to drive our Realm Server DOM tests. Details here: https://developer.chrome.com/blog/immutable-document-domain/

This PR refactors our realm server DOM tests to leverage `postMessage` to communicate with the Realm's document that resides within an iframe.

### CODE REVIEW

File: patches/http-server@14.1.1.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/lib/core/index.js b/lib/core/index.js
+index 920e55b9c01222c280b7f629ab4c8aaac8a987f7..a2464e449d59cf184ff89ac53f048639b541310e 100644
+--- a/lib/core/index.js
++++ b/lib/core/index.js
+@@ -319,6 +319,8 @@ module.exports = function createMiddleware(_dir, _options) {
+       res.setHeader('content-length', stat.size);
+       res.setHeader('content-type', contentType);
+ 
++      res.setHeader('Origin-Agent-Cluster', '?0');
```
Comment (habdelra) [inline patches/http-server@14.1.1.patch:9]: we tweak the http-server that we use to drive the tests to permit it to consider the localhost on a different realm as part of the same origin. This is the new way that chrome expects you to allow that. Despite doing this, you still cannot directly manipulate the iframe's document (this just eliminates one barrier around that), hence the `postMessage` communication in the realm-dom-test.js

**Comment by habdelra** [comment]
arg, the tests hang because testem is not actually serving the test container with the special new header value...

**Review by lukemelia — APPROVED** [review]

File: packages/realm-server/dom-tests/realm-dom-test.js [hunk 2]
```diff
@@ -30,26 +72,62 @@ async function waitFor(selector, timeoutMs = 10000) {
...
+async function querySelector(selector, messenger) {
+  let fragment = await messenger.send(
+    { querySelector: selector },
+    testRealmURL,
+  );
+  if (fragment == null) {
+    return null;
+  }
+  attachFragments(fragment);
+  return document.querySelector(`#${iframeSelectorTempId} > ${selector}`);
```
Comment (habdelra) [inline packages/realm-server/dom-tests/realm-dom-test.js]: Should only query with the leaf-most part of the selector

File: packages/realm-server/dom-tests/realm-dom-test.js [hunk 3]
```diff
@@ -30,26 +72,62 @@ async function waitFor(selector, timeoutMs = 10000) {
...
-function querySelectorAll(selector) {
-  let doc = testDocument();
-  return doc?.querySelectorAll(selector);
+async function querySelectorAll(selector, messenger) {
+  let fragments = await messenger.send(
+    { querySelectorAll: selector },
+    testRealmURL,
+  );
+  attachFragments(fragments);
+  return document.querySelectorAll(`#${iframeSelectorTempId} > ${selector}`);
```
Comment (habdelra) [inline packages/realm-server/dom-tests/realm-dom-test.js]: Should only query with the leaf-most part of the selector

**Review by jurgenwerk — APPROVED** [review]

**Pull request merged**